python/lsst/obs/subaru/isr.py

```diff
@@ -307,24 +337,23 @@ def runDataRef(self, sensorRef):
             self.darkCorrection(ccdExposure, darkExposure)
         if self.config.doFlat:
             flatExposure = self.getIsrExposure(sensorRef, "flat")
-            self.flatCorrection(ccdExposure, flatExposure)
+            self.flatCorrection(ccdExposure, flatExposure, doTweakFlat=self.config.doTweakFlat)
 
-        if self.config.doDefect:
+        if self.config.doDefect and not interpolationDone:
             self.maskAndInterpDefect(ccdExposure, defects)
 
         if self.config.doApplyGains:
             self.applyGains(ccdExposure, self.config.normalizeGains)
-        if self.config.doWidenSaturationTrails:
-            self.widenSaturationTrails(ccdExposure.getMaskedImage().getMask())
-        if self.config.doSaturation:
+        if self.config.doSaturation and not interpolationDone:
             self.saturationInterpolation(ccdExposure)
 
         if self.config.doFringe:
             self.fringe.runDataRef(ccdExposure, sensorRef)
         if self.config.doSetBadRegions:
             self.setBadRegions(ccdExposure)
 
-        self.maskAndInterpNan(ccdExposure)
+        if not interpolationDone:
+            self.maskAndInterpNan(ccdExposure)
```

Comment:
I think we want this to fire always --- there might be zeros in the flat.

Reply:
I'm planning to go with @RobertLuptonTheGood's proposal to make this a config option but disable it by default.